internal/pkg/flink/app/application.go

```diff
@@ -58,7 +58,7 @@ func StartApp(client ccloudv2.GatewayClientInterface, tokenRefreshFunc func() er
 	// Instantiate Component Controllers
 	inputController := controller.NewInputController(historyStore)
 	statementController := controller.NewStatementController(appController, dataStore, consoleParser)
-	interactiveOutputController := controller.NewInteractiveOutputController(components.NewTableView(), resultFetcher, appOptions.GetVerbose())
+	interactiveOutputController := controller.NewInteractiveOutputController(components.NewTableView(), resultFetcher, reportUsageFunc, appOptions.GetVerbose())
 	basicOutputController := controller.NewBasicOutputController(resultFetcher, inputController.GetWindowWidth)
 
 	app := Application{
```

Comment:
Panic recovery for interactive output works great 👍 

a.statementController.ExecuteStatement(userInput), in the input mode, can however still panic leading to statements not being deleted before the results were fetched. I think we should have some panic recovery for that case, so we delete the statement if the shell panics while the statement is being created/waiting to be ready/we're listening for errors.

Reply:
Yeah that makes sense! I've refactored it now, so that the panic recovery happens in the main app loop. That way, regardless of where it panics, if we created a statement it will get deleted.

Reply:
nice!